docs/.linkcheckerrc

```diff
@@ -9,6 +9,7 @@ threads=1
 ignore=
   pythonapi/lightgbm\..*\.html.*
   http.*amd.com/.*
+  https.*tandfonline.com/.*
```

Comment:
I'm able to navigate to https://www.tandfonline.com/doi/abs/10.1080/01621459.1958.10501479 in my browser without issue, and without being signed in to that website.

<img width="1421" alt="image" src="https://github.com/microsoft/LightGBM/assets/7608904/3e9f5c23-b297-4b16-b6ee-a937e1e350f7">

But `curl -I` (which issues a `HEAD` request), returns a `403 - Access Denied`.

```text
(lgb-dev) 😬 ./LightGBM $ curl -I https://www.tandfonline.com/doi/abs/10.1080/01621459.1958.10501479
HTTP/2 403
date: Fri, 23 Jun 2023 04:59:22 GMT
content-type: text/html; charset=UTF-8
cross-origin-embedder-policy: require-corp
cross-origin-opener-policy: same-origin
cross-origin-resource-policy: same-origin
permissions-policy: accelerometer=(),autoplay=(),camera=(),clipboard-read=(),clipboard-write=(),geolocation=(),gyroscope=(),hid=(),interest-cohort=(),magnetometer=(),microphone=(),payment=(),publickey-credentials-get=(),screen-wake-lock=(),serial=(),sync-xhr=(),usb=()
referrer-policy: same-origin
x-frame-options: SAMEORIGIN
cf-mitigated: challenge
cache-control: private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0
expires: Thu, 01 Jan 1970 00:00:01 GMT
strict-transport-security: max-age=15552000
server: cloudflare
cf-ray: 7dba29492ca810b3-ORD
alt-svc: h3=":443"; ma=86400
```

Looks like that site is behind CloudFlare, so the link checks are probably triggering some anti-DDoS protection 🤷🏻 

This configuration change tells the `check-links` CI job to ignore that domain.

A similar issue with `amd.com` links was fixed by doing this same thing in the past here:

* #3529
* #3530